src/components/main.tsx

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import classNames from 'classnames';
+import { FaRegEdit } from 'react-icons/fa';
+import React, { useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  data:Todo[];
+  setData:(data: Todo[]) => void;
+};
+
+export const Main: React.FC <Props> = ({ data, setData }) => {
+  const [editTodo, setEditTodo] = useState(0);
+  const [editValue, setEditValue] = useState('');
+  const [isToggleAll, setIsToggleAll] = useState(false);
+
+  const handleRemove = (id:number) => {
+    const newList = data.filter((item) => item.id !== id);
+
+    setData(newList);
+  };
+
+  const handleCheck = (id:number): void => {
+    setData(data.map(todo => {
+      if (todo.id === id) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+
+      return todo;
+    }));
+  };
+
+  const handleToggleAll = () => {
+    setIsToggleAll(!isToggleAll);
+
+    if (isToggleAll) {
+      setData(data.map(todo => {
+        return {
+          ...todo,
+          completed: false,
+        };
+      }));
+    }
+
+    if (!isToggleAll) {
+      setData(data.map(todo => {
+        return {
+          ...todo,
+          completed: true,
+        };
+      }));
+    }
+  };
+
+  const handleEdit = (id:number, title: string) => {
+    setEditTodo(id);
+    setEditValue(title);
+  };
+
+  // save changes onBlur
+  const handleSubmitValue = (id:number) => {
+    const newValue = data.map(todo => {
+      if (todo.id === id) {
+        todo.title = editValue;
+      }
+
+      return todo;
+    });
+
+    setData(newValue);
+    setEditTodo(0);
+
+    data.map(todo => {
+      if (!todo.title) {
+        handleRemove(id);
+      }
+
+      return todo;
+    });
+  };
+
+  // save changes with Enter press
+  const handleEnter = (e:any, id:number) => {
+    if (e.code === 'Enter') {
+      const newValue = data.map(todo => {
+        if (todo.id === id) {
+          todo.title = editValue;
```

Comment:
```suggestion
         return { ...todo, title: editValue }
```
в мапі ж об'єкт вертаємо, не асайнимо в поля нові значення на пряму